src/Interpreters/Cache/FileSegment.cpp

```diff
@@ -157,6 +157,24 @@ size_t FileSegment::getReservedSize() const
     return reserved_size;
 }
 
+size_t FileSegment::getSize(FileSegment::SizeAlignment alignment) const
+{
+    size_t size = getReservedSize();
+    switch (alignment)
+    {
+        case FileSegment::SizeAlignment::ALIGNED:
+            return getKeyMetadata()->alignFileSize(size);
+        case FileSegment::SizeAlignment::NOT_ALIGNED:
+            return size;
+        case FileSegment::SizeAlignment::DEFAULT_ALIGNMENT:
+            auto locked_key = getKeyMetadata();
+            chassert(locked_key);
+            return locked_key->useRealDiskSize() ? locked_key->alignFileSize(size) : size;
```

Comment:
This is not a "locked_key" (it would have been so if we did `getKeyMetadata().lock()`), so let's call it differently